### ISSUE TITLE
Grid Paste option does not exist in GUI

### ISSUE THREAD

**Issue by nice6599** [maruohon/litematica#177]
The grid paste option does not exist in the fabric version of this mod. There is a Grid Paste Setting for the tool but no hotkeys as there should be.

**Comment by vacla** [comment]
it is already implemented in 1.12.2, it will be implemented later when the next merge happens

**Comment by maruohon** [comment]
For reference, the activation hotkey for Grid Pasting is the same `executeOperation` key as the other tool actions use. And to control the grid settings, there is a new button in the placement configuration GUI and also a new hotkey to quickly open that GUI directly.

**Comment by nice6599** [comment]
awesome

**Comment by maxsupermanhd** [comment]
@maruohon any merge plans? I'm running 1.16

**Comment by maruohon** [comment]
Plans: yes of course.
Timeline/ETA: no.
Soon: not very.

I'm still finishing up the malilib rewrite on 1.12.2, after which I'll start working on the other mods towards the next big update cycle. No idea how long that will take. And there are 4 of those other mods to work on (Litematica, Item Scroller, MiniHUD and Tweakeroo). Plus I also have a new mod planned/started, which also kinda needs to be done before that new malilib is released publicly, in case it would need some changes still to malilib.

**Comment by Bluscream** [comment]
Any news?

**Comment by maruohon** [comment]
Unfortunately no news to report still... Progress on the main development code has been very slow if non-existent for some time now. But I'm trying to get back to working on the remaining refactors on 1.12.2 after I get the rest of the mods out for 1.18.

**Comment by KnightYoshi** [comment]
So we're a year and a half since the last update, any updates for this? 👀

**Comment by maruohon** [comment]
Not much to report still... Still working on the 1.12.2 code in preparation for some initial "test ports" to more recent MC versions. malilib is getting rather close to being ready for first "proper public release", but Litematica still has several areas of code that need to be rewritten and cleaned up before I'll port it to any other MC versions.

**Comment by goldengun111** [comment]
Any 2025 updates on this?

**Comment by maruohon** [comment]
Not much to report. I've barely done any modding in the past couple of years. Although I have been getting back to it a little bit lately, but still just slowly working on the rewrites in 1.12.2.

**Comment by Victrecated** [comment]
It's probably quite annoying to get a consistent ask on updates.
But any updates for 2026? I was hoping to use the grid paste in 1.21.04, but it seems like Fabric is a bit broken currently...

**Comment by maruohon** [comment]
Nope, same situation as before: I only get a little bit of stuff done mostly during summer and Christmas holidays these days. My summer holiday starts next week, so let's see if I manage to make any progress on anything during it this year...

But still the main focus is in 1.12.2, and it won't get ported to the modern MC versions before I finish most of the planned changes. And there is no telling how many years that will take with this slow pace...
